python/ribasim/ribasim/input_base.py

```diff
@@ -202,6 +216,44 @@ def node_ids(self) -> set[int]:
 
         return node_ids
 
+    def offset_node_ids(self, offset_node_id: int) -> "TableModel[TableT]":
+        """Add the same offset to all node IDs."""
+        copy = deepcopy(self)
+        df = copy.df
+        if isinstance(df, (pd.DataFrame, gpd.GeoDataFrame)):
+            df.index += offset_node_id
+            for name_column in [
+                "node_id",
+                "from_node_id",
+                "to_node_id",
+                "listen_feature_id",
+            ]:
+                if hasattr(df, name_column):
+                    df[name_column] += offset_node_id
+        return copy
+
+    def merge_table(
```

Comment:
I'd expected this function to also fix any index/id clashes by calling some form of offset?

Reply:
I've added a check that the tables have no node IDs in common.